### PR TITLE
Jason Added Staff Index Page (Redirected from a Course's Staff Button)

### PR DESCRIPTION
**Pull request by jason-rodrigues**
Updated Staff Table Loading:
image
Edit and Create Page Redirecting to Placeholders (I guessed for some of the URLs... if I should change these, lmk)
image
image

Deployed on Dokku [here]( https://proj-organic-jason-rodrigues-dev.dokku-12.cs.ucsb.edu) 
(Staff + Team should be admin, lmk if there's any issues)

### CODE REVIEW

**Review by ElijahCanderson — DISMISSED** [review]
LGTM

File: frontend/src/App.js [hunk 1]
```diff
@@ -42,6 +45,9 @@ function App() {
       <Route path="/courses/create" element={<CoursesCreatePage />} />
       <Route path="/courses" element={<CourseIndexPage />} />
       <Route path="/courses/edit/:id" element={<CoursesEditPage />} />
+      <Route path="/courses/:id/staff" element={<CoursesStaffPage />} />
+      <Route path="/courses/:id/editStaff" element={<StaffEditPage />} />
```
Comment (pconrad) [inline frontend/src/App.js]:
```suggestion
```

There is no use case for a staff edit button.  It's only add and delete.

File: frontend/src/App.js [hunk 2]
```diff
@@ -15,9 +15,12 @@ import SchoolIndexPage from "main/pages/SchoolIndexPage";
 
 import CoursesCreatePage from "main/pages/CoursesCreatePage";
 import CourseIndexPage from "main/pages/CourseIndexPage";
+import CoursesStaffPage from "main/pages/CoursesStaffPage";
 
 import { hasRole, useCurrentUser } from "main/utils/currentUser";
 import NotFoundPage from "main/pages/NotFoundPage";
+import StaffEditPage from "main/pages/StaffEditPage";
```
Comment (pconrad) [inline frontend/src/App.js]: There is no use case for a staff edit button. It's only add and delete.

File: frontend/src/main/components/Staff/StaffTable.js [hunk 3]
```diff
@@ -22,8 +41,15 @@ import OurTable from "main/components/OurTable"
    
      ];
 
-     return <OurTable
-         data={staff}
-         columns={columns}
-         testid={"StaffTable"} />;
+     if (hasRole(currentUser, "ROLE_ADMIN") || hasRole(currentUser, "ROLE_INSTRUCTOR")) {
+        columns.push(ButtonColumn("Edit", "primary", editCallback, "StaffTable"));
```
Comment (pconrad) [inline frontend/src/main/components/Staff/StaffTable.js]: There is no use case for a staff edit button. It's only add and delete.
```suggestion
```

File: frontend/src/main/pages/StaffEditPage.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+  // Stryker disable all : placeholder for future implementation
+  return (
+    <BasicLayout>
+      <div className="pt-2">
+        <h1>Edit page not yet implemented</h1>
+      </div>
+    </BasicLayout>
+  )
+}
```
Comment (pconrad) [inline frontend/src/main/pages/StaffEditPage.js]: Remove file: There is no use case for a staff edit button. It's only add and delete.

File: frontend/src/stories/pages/StaffEditPage.stories.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        rest.put('/api/helprequest', async (req, res, ctx) => {
+            var reqBody = await req.text();
+            window.alert("PUT: " + req.url + " and body: " + reqBody);
+            return res(ctx.status(200),ctx.json({}));
+        }),
+    ],
+}
+
+
+
```
Comment (pconrad) [inline frontend/src/stories/pages/StaffEditPage.stories.js]: There is no use case for a staff edit button. It's only add and delete.

File: frontend/src/tests/pages/StaffEditPage.test.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import StaffEditPage from "main/pages/StaffEditPage";
+import { QueryClient, QueryClientProvider } from "react-query";
+import { MemoryRouter } from "react-router-dom";
+
+import { apiCurrentUserFixtures } from "fixtures/currentUserFixtures";
+import { systemInfoFixtures } from "fixtures/systemInfoFixtures";
+import axios from "axios";
+import AxiosMockAdapter from "axios-mock-adapter";
+
+
```
Comment (pconrad) [inline frontend/src/tests/pages/StaffEditPage.test.js]: There is no use case for a staff edit button. It's only add and delete.

**Review by pconrad — CHANGES_REQUESTED** [review]
Overall, code looks good!  

But we need to remove edit; there's literally nothing to edit for a staff member.  A github user is either a member of the staff for a given course, or they are not.  So it's just add or delete; nothing else.

**Review by Division7 — DISMISSED** [review]
LGTM 😎 🤙

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**